algebird-test/src/test/scala/com/twitter/algebird/maps/mixed.scala

```diff
@@ -0,0 +1,163 @@
+/*
+Copyright 2012 Twitter, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package com.twitter.algebird.maps
+
+import org.scalatest._
+
+import com.twitter.algebird.Monoid
+
+import com.twitter.algebird.matchers.seq._
+
+object mixed {
+  import math.Numeric
+  import scala.collection.SortedMap
+
+  import com.twitter.algebird.maps.increment._
+  import com.twitter.algebird.maps.prefixsum._
+  import com.twitter.algebird.maps.nearest._
+
+  object tree {
+    import com.twitter.algebird.maps.increment.tree._
+    import com.twitter.algebird.maps.prefixsum.tree._
+    import com.twitter.algebird.maps.nearest.tree._
+
+    trait NodeMix[K, V, P] extends NodePS[K, V, P] with NodeInc[K, V] with NodeNearMap[K, V]
+
+    trait LNodeMix[K, V, P] extends NodeMix[K, V, P]
+      with LNodePS[K, V, P] with LNodeInc[K, V] with LNodeNearMap[K, V]
+
+    trait INodeMix[K, V, P] extends NodeMix[K, V, P]
+      with INodePS[K, V, P] with INodeInc[K, V] with INodeNearMap[K, V] {
+      def lsub: NodeMix[K, V, P]
+      def rsub: NodeMix[K, V, P]
+    }
+  }
+
+  import tree._
+
+  object infra {
+    import com.twitter.algebird.maps.redblack.tree._
+    import com.twitter.algebird.maps.ordered.tree.DataMap
+
+    class Inject[K, V, P](
+      val keyOrdering: Numeric[K],
```

Comment:
`keyOrdering` is just the general ordering relation for all keys, and in general it is only required to be `Ordering`.   However, in the case of the subclasses that provide nearest-neighbor queries, keys must also support a notion of distance, defined as `|k1 - k2|`, and more generically obey the idea that when `k1 < k2 < k3`, then also `k2 - k1 < k3 - k1`.

In other words, for the nearest-neighbor subclass, the keys aren't just ordered, they are numeric.   This works out nice and clean, since `Numeric <: Ordering` in scals's type system, so that subclass just tightens the type constraint to `Numeric`